### PR TITLE
Updated data scraper service (websites, reports)

### PR DESCRIPTION
**Pull request by skhan943**
### JIRA Ticket
https://skaarkzo-capstone.atlassian.net/browse/SUSTAIN-15

### Summary of Changes
- Added big list of Canadian companies to help with website discovery service
- Updated scraping service, big improvements in content fetching and data structure
- Added ability to download and extract content from PDFs
- Abstracted PDF processing

### CODE REVIEW

**Review by rakhshankhan — COMMENTED** [review]
There doesn't seem to be an endpoint that's returning the scraped data. Would be worth adding it as part of this MR.

**Review by rakhshankhan — COMMENTED** [review]
After Zaid's implementation is done, I've proposed a new way to organize stuff in the data scraper since it feels a bit confusing rn. Any scrapers should go under a scapers folder under service (i.e. company_website_scraper, news_article_scraper, etc.). We can remove utilities folder from now since primary functionality of pdf_processor is to scrape PDFs, so move that under scraper too as pdf_scraper.

Any automations will go under an automations folder under service (i.e. sedar_automation, etc.).

Root service folder will only have service files that contain methods referenced by some controller.

Lmk if any objections or better ideas.

**Comment by zaidokal** [comment]
> After Zaid's implementation is done, I've proposed a new way to organize stuff in the data scraper since it feels a bit confusing rn. Any scrapers should go under a scapers folder under service (i.e. company_website_scraper, news_article_scraper, etc.). We can remove utilities folder from now since primary functionality of pdf_processor is to scrape PDFs, so move that under scraper too as pdf_scraper.
> 
> Any automations will go under an automations folder under service (i.e. sedar_automation, etc.).
> 
> Root service folder will only have service files that contain methods referenced by some controller.
> 
> Lmk if any objections or better ideas.

All restructuring have been completed in this MR: https://github.com/skaarkzo-capstone/data-scraper-service/pull/7

File: app/dto/requests/process_request_dto.py
Comment (rakhshankhan) [inline app/dto/requests/process_request_dto.py:1]: Let's quickly rename this to process_request. DTO is only used for response objects I believe. Also folder name should be singular >:)

**Comment by zaidokal** [comment]
> There doesn't seem to be an endpoint that's returning the scraped data. Would be worth adding it as part of this MR.

Endpoints have been added in this MR: https://github.com/skaarkzo-capstone/data-scraper-service/pull/8
And is now in this branch.

File: app/scraper/company_website_scraper.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+        # Keywords to exclude certain pages (e.g., login, careers, etc.)
+        self.exclusion_keywords = [
+            "login", "sign in", "sign_in", "signin", "sign-in",
+            "privacy policy", "terms", "terms of service", "terms-of-service",
+            "careers", "jobs", "employment",
+            "support", "help", "faq", "documentation", "customer service",
+            "contact us", "contactus", "contact-us", "blog", "newsletter",
+            "what's new"
+        ]
```
Comment (rakhshankhan) [inline app/scraper/company_website_scraper.py:55]: Separate util file for these would be nice.

Comment (zaidokal) [inline app/scraper/company_website_scraper.py:55]: Can you move these to a different file?

**Review by rakhshankhan — APPROVED** [review]

**Review by zaidokal — APPROVED** [review]

**Pull request merged**